app/vmagent/remotewrite/pendingseries.go

```diff
@@ -241,7 +274,7 @@ func (wr *writeRequest) copyTimeSeries(dst, src *prompbmarshal.TimeSeries) {
 var marshalConcurrencyCh = make(chan struct{}, cgroup.AvailableCPUs())
 
 func tryPushWriteRequest(wr *prompbmarshal.WriteRequest, tryPushBlock func(block []byte) bool, isVMRemoteWrite bool) bool {
-	if len(wr.Timeseries) == 0 {
+	if len(wr.Timeseries) == 0 && len(wr.Metadata) == 0 {
```

Comment:
I'd suggest to add `wr.isEmpty` check to perform both checks. That will help to ensure it is properly checked everywhere and will make it easier to add more items to `wr` (maybe exemplars some time in future).